### DISCUSSION TITLE
v2.1.0

### DISCUSSION THREAD

**Discussion by wanghan-iapcm** [deepmodeling/deepmd-kit#1544] [Announcement]
New feature:
- Model compression for se_3, se_r descriptors. Energy and tensor models (#1225 #1228 #1361  )
- Add init-frz-model support for se-t type descriptor (#1245)
- Added all activation functions for model compression. (#1283)
- Update guidelines for the number of threads (#1291)
- Enable mixed precision support for deepmd-kit (#1285 #1471 )
- Unify C++ errors and pass message to LAMMPS (#1326)
- Optimize `DPTabulate._build_lower` method (#1323)
- Calculate neighbor statistics from CLI (#1476)
- Add an interface to eval descriptors (#1483)

Enhancement:
- deprecate `numb_test` in the training script (#1249)
- Accelerate model compression (#1274)
- Use c++14 for TF 2.7 (#1275)
- Add a citation badge (#1280)
- Add embedding network dimension check of model compression (#1303)
- Provide an option to skip neighbor stat (#1313)
- Add an error message to compress/freeze (#1319)
- Redirect `print_summary` to LAMMPS log (#1324)
- Enable OpenMP for `prod_force` and `prod_virial` (#1360)
- Update issue templates (#1368)
- Bump LAMMPS version to `stable_29Sep2021_update2` (#1279)
- Remove `api_cc/include/custom_op.h` (#1405)
- Introduce TensorFlow Profiler (#1414)
- Only test/eval fitting properties during training (#1416 #1419 )
- Remove the dependency on `inputs` from `inputs_zero` (#1417)
- Support recursive detection for the systems of model_devi (#1424)
- Enable TF remapper optimizer (#1418)
- Dynamically load op library in C++ interface (#1384)
- Dplr doc and examples (#1458)
- Bump the Python version to 3.10 (#1465)
- Do some small optimization to ops (avoid concat or add in loops. Instead, append tensors to a list, and concat or accumulate_n after loops) (#943)
- Optimizations related to data statistics
    - Skip `data_stat` in `init_from_model` and `restart` mode (#1463)
    - Assign energy shift stats if atomic energies are assigned (#1477)
    - Recover input stats from frozen models (#1482)
- Test: move loading graphs to setUpClass to accelerate tests (#1484)
- Run test_python in the pre-built container (#1487)

Bug fixings:
- Update and fix typos in doc (#1238 #1239 #1328  #1300 #1445 #1490 #1497 #1504 #1503 #1514 )
- Fix compress training bug within the dp train --init-frz-model interface (#1233)
- Fix Python bugs of loc_frame descriptor (#1253)
- Fix bug of loc_frame descriptor when using lammps (#1255)
- Fix single precision error (#1212)
- Fix the np.frombuffer in dp transfer (#1246)
- Fix SyntaxWarning in graph.py (#1278)
- Change googletest from master to main (#1292)
- `update_deepmd_input` when compress (#1297 #1301 )
- Add importlib_metadata as dependency (#1308)
- Fix bugs about parameters of memset (#1302)
- Fix model compression bug when fparam or aparam is not zero (#1306)
- Add space between words in messages (#1312)
- Do not print virial error with nopbc data (#1314)
- Fix test errors with TensorFlow 2.7 (#1315)
- Fix bug of hip model compression (#1325)
- Prevents rcut_smth larger than rcut (#1354)
- Fix cell and virial transpose bug in dp_ipi (#1353)
- Fix bug in DipoleFittingSeA: (#1363)
- Fix cxx standard for LAMMPS (#1379)
- Explicitly set neighbor request to full in compute deeptensor/atom to fix bug #1381 (#1382)
- Fix NameError (#1385)
- Fix network precision under specific situation (#1391 #1394 )
- Initialize input virial vector to zero (#1397)
- Make OpenMP an optional dependency (#1498)
- Fix nvcc warning when using cuda-11.x toolkit (#1401)
- Add UT for se_3 type descriptor (#1404)
- Fix github git url (#1409)
- Fix gelu grad multi definitions error (#1406)
- Fix `cast_precision` breaking docstring (#1437)
- Add image link of ROCm version. (#1432) 
- Pass integer zero to memset (#1499)

<hr /><em>This discussion was created from the release <a href='https://github.com/deepmodeling/deepmd-kit/releases/tag/v2.1.0'>v2.1.0</a>.</em>

**Comment by likefallwind** [comment]
Cool！
